specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2022-01-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,1096 @@
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration S3 bucket.",
+          "type": "string"
+        },
+        "bucketName": {
+          "description": "The bucket name to sync from the url endpoint for the flux configuration.",
+          "type": "string"
+        },
+        "insecure": {
+          "description": "Specify whether to use insecure communication when puling data from the S3 bucket.",
+          "type": "boolean",
+          "default": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "accessKey": {
+          "description": "Plaintext access key used to securely access the S3 bucket",
+          "type": "string",
+          "x-nullable": true
+        },
+        "localAuthRef": {
+          "description": "Name of a local secret on the Kubernetes cluster to use as the authentication secret rather than the managed or user-provided configuration secrets.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "BucketPatchDefinition": {
+      "description": "Parameters to reconcile to the GitRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration S3 bucket.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "bucketName": {
+          "description": "The bucket name to sync from the url endpoint for the flux configuration.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "insecure": {
+          "description": "Specify whether to use insecure communication when puling data from the S3 bucket.",
+          "type": "boolean",
+          "x-nullable": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "x-nullable": true
+        },
+        "accessKey": {
```

Comment:
same x-ms-secret comment

Reply:
This is not a secret, this is similar to username for an S3 bucket